alibi/explainers/pd_variance.py

```diff
@@ -0,0 +1,660 @@
+            buffers = self._compute_feature_interaction(pd_explanation=pd_explanation,
+                                                        features=features)  # type: ignore[arg-type]
+
+        # update meta and return built explanation
+        self.meta['params'].update(self.pd_explainer.meta['params'])
+        self.meta['params'].update({'method': method})
+        return self._build_explanation(buffers=buffers)
+
+    def _compute_pd_variance(self, features: List[int], pd_values: List[np.ndarray]) -> np.ndarray:
+        """
+        Computes the PD variance along the final axis for all the features.
+
+        Parameters
+        ----------
+        features
+            See :py:meth:`alibi.explainers.pd_variance.PartialDependenceVariance.explain`.
+        pd_values
+            List of length `F` containing the PD values for each feature in `features`. Each PD value is an array
+            with the shape `T x N1 ... x Nk` where `T` is the number of targets and `Ni` is the number of feature
+            values along the `i` axis.
+
+        Returns
+        -------
+        An array of size `F x T x N1 x ... N(k-1)`, where `F` is the number of explained features, `T` is the number
+        of targets, and `Ni` is the number of feature values along the `i` axis.
+        """
+        feature_variance = []
+
+        for pdv, f in zip(pd_values, features):
+            # `pdv` is a tensor of size `T x N1 x ... Nk`, where `T` is the number
+            # of targets and `Ni` is the number of feature values along the axis `i`
+            if f in self.pd_explainer.categorical_names:  # type: ignore[operator]
+                ft_var = PartialDependenceVariance._compute_pd_variance_cat(pdv)
+            else:
+                ft_var = PartialDependenceVariance._compute_pd_variance_num(pdv)
+            # add extra dimension for later concatenation along the axis 0
+            feature_variance.append(ft_var[None, ...])  # type: ignore[index]
+
+        # stack the feature importance such that the array has the shape `F x T x N1 ... N(k-1)`
+        return np.concatenate(feature_variance, axis=0)
+
+    @staticmethod
+    def _compute_pd_variance_num(pd_values: np.ndarray) -> np.ndarray:
+        """
+        Computes the PD variance along the final axis for a numerical feature.
+
+        Parameters
+        ----------
+        pd_values
+            Array of partial dependence values for a numerical features of size `T x N1 x ... x Nk`, where `T` is the
+            number of targets and `Ni` is the number of feature values along the axis `i`.
+
+        Returns
+        -------
+        PD variance along the final axis for a numerical feature.
+        """
+        return np.std(pd_values, axis=-1, ddof=1, keepdims=False)
+
+    @staticmethod
+    def _compute_pd_variance_cat(pd_values: np.ndarray) -> np.ndarray:
+        """
+        Computes the PD variance along the final axis for a categorical feature.
```

Comment:
Would be good to leave a note to a reference for this "range statistic" heuristic so people unfamiliar with it looking at the code later on are not confused.